web/src/engine/websites/LezhinBase.ts

```diff
@@ -0,0 +1,381 @@
+                });
+            return Promise.resolve(chapters);
+        };
+
+        const request = this.createRequest(new URL(`/comic/${manga.Identifier}`, this.URI));
+        let chapters = await FetchWindowScript(request, script, 2500, 10000);
+        return chapters.map((chap) => new Chapter(this, manga, chap.name,
+            `${chap.display.displayName} - ${chap.display.title}`));
+    }
+
+    public async FetchPages(chapter: Chapter): Promise<Page[]> {
+        await this.initializeAccount();
+
+        const script = function(this: Window, chapterName: string): Promise<boolean> {
+            const product = (this as LezhinWindow).__LZ_PRODUCT__ ?? (this as LezhinWindow).__LZ_DATA__;
+            if (!product) {
+                return Promise.reject('Could not get chapters product data (LezhinBase:FetchPages). Available keys are: '
+                    + Object.keys(this.window).filter(a => (a > '__LZ^' && a < '__LZ`')).join(', '));
+            }
+            const chapter = product.all.find(chapter => chapter.name == chapterName);
+            return Promise.resolve(chapter.purchased);
+        };
+
+        const scriptText = `(${script})(
+            ${this.stringToLiteral(chapter.Identifier)}
+        )`;
+
+        const purchasedRequest = this.createRequest(new URL(`/comic/${chapter.Parent.Identifier}`, this.URI));
+        let purchased: boolean = await FetchWindowScript(purchasedRequest, scriptText, 2500, 10000);
+
+        const uri = new URL('/lz-api/v2/inventory_groups/comic_viewer', this.apiURI);
+        const params = new URLSearchParams ({
+            'platform': 'web',
+            'store': 'web',
+            'alias': chapter.Parent.Identifier,
+            'name': chapter.Identifier,
+            'preload': 'false',
+            'type': 'comic_episode'
+        });
+        uri.search = params.toString();
+        const request = this.createRequest(uri);
+        const content = await FetchJSON<APIPages>(request);
+
+        const parameters: EpisodeParameters = {
+            episodeID: content.data.extra.episode.id,
+            comicID: content.data.extra.episode.idComic,
+            updatedAt: content.data.extra.episode.updatedAt,
+            shuffled: content.data.extra.comic.metadata?.imageShuffle ?? false,
+            purchased: purchased,
+            subscribed: content.data.extra.subscribed,
+        };
+        const extension = this.Settings.forceJPEG.Value ? '.jpg' : '.webp';
+
+        const pages = content.data.extra.episode.pagesInfo ?? content.data.extra.episode.scrollsInfo;
+        return pages.map((page) => {
+            const url = new URL('/v2' + page.path + extension, this.cdnURI);
+            return new Page(this, chapter, url, parameters);
+        });
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        const parameters = page.Parameters as EpisodeParameters;
```

Comment:
In FetchImages, fetch request should be wrapped in task unless they is only one and its FetchImageAjax.

https://github.com/manga-download/haruneko/blob/51752d580920db0a77276f17092f7e3dde6e5b45/web/src/engine/websites/Mangadon.ts#L92